components/wpt-results.html

```diff
@@ -22,20 +22,94 @@
     <style>
       :host {
         display: block;
+        font-size: 16px;
       }
       .spec {
         background-color: #eee;
         padding: 1em;
         margin-bottom: 1em;
         border: solid 1px #ccc;
       }
+      section.search {
+        border-bottom: solid 1px #ccc;
+        padding-bottom: 1em;
+        margin-bottom: 1em;
+      }
+      input.query {
+        font-size: 16px;
+        display: block;
+        padding: 0.5em 0;
+        width: 100%;
+      }
+      .browsers.headers {
+        padding: 1em;
+      }
+      .browsers {
+        display: flex;
+        flex-direction: row-reverse;
+      }
+      .browser {
+        display: inline-block;
+        font-size: 0.8em;
+        min-width: 100px;
+      }
+      .browser img {
+        width: 32px;
+        height: 32px;
+      }
     </style>
 
-    <template is="dom-repeat" items="{{specs}}">
+    <section class="search">
+      <input
+        value="{{query::input}}"
+        class="query"
+        placeholder="Search test files, like cors/allow-headers.htm">
+    </section>
+
+    <section class="browsers headers">
+      <template is="dom-repeat" items="{{testRuns}}">
+        <div class="browser">
+          <div><img src="/static/{{item.browser_name}}_64x64.png" /></div>
+          <div>{{item.browser_name}} {{item.browser_version}}</div>
+          <div>{{item.os_name}} {{item.os_version}}</div>
+          <div>@{{item.revision}}</div>
+        </div>
+      </template>
+    </section>
+
+    <template
+      is="dom-repeat"
+      items="{{specs}}" as="spec" filter="_specFilter" id="spec_list" sort="_specSort">
       <div class="spec">
-        <b>{{item.spec}}</b><br/>
-        {{item.passing}} / {{item.total}}<br/>
-        {{_passingPercent(item)}}%
+        <div>
+          <b style="display: inline-block;">{{spec.specName}}</b>
+
+          <div class="browsers">
+            <template is="dom-repeat" items="{{spec.results}}" as="result">
+              <div class="browser">
+                <div class="browser">{{ result.passing }} / {{ result.total }}<br>{{ _passingPercent(result) }}%</div>
+              </div>
+            </template>
+          </div>
+        </div>
+
+        <!-- TODO(jeffcarp): This nested sort isn't working -->
+        <template is="dom-repeat" items="{{spec.testFiles}}" as="testFile" filter="_testFileFilter" sort="_testFileSort" class="test_file_list">
+          <div>{{ testFile.testFile }}</div>
+
+          <!-- This is the only way I've found to get sub-lists to update -->
+          <div style="display:none;">{{query}}</div>
```

Comment:
Including `query` in the subtest result sub-tree is the only way I could get the subtest results to update each time `query` is updated. I'll ask someone on the Polymer team if there's a more idiomatic way to do this.